### PR TITLE
Ensure that only enabled modules react to queries

### PR DESCRIPTION
**Pull request by adnasa**
Addresses #64

### CODE REVIEW

File: app/main/plugins.js [hunk 1]
```diff
@@ -5,28 +5,53 @@ const plist = require('plist');
...
+const readConfig = () => (new Promise((resolve, reject) => {
+  const file = `${DEXT_PATH}/config.json`;
+  fs.readFile(file, (err, data = '{}') => {
+    if (err) {
+      reject(err);
+      return;
+    }
+    resolve(JSON.parse(data));
+  });
+}));
```
Comment (adnasa) [inline app/main/plugins.js]: hmm, I just realised that I probably should `dext-core-utils` here,
correct, @vutran ?

Comment (vutran) [inline app/main/plugins.js]: Yes, this function should be moved to `dext-core-utils` since we'll most likely be reusing it on `dext-cli`.

Comment (adnasa) [inline app/main/plugins.js]: Will fix that then ;)

Comment (adnasa) [inline app/main/plugins.js]: @vutran I would want to suggest the push usage of `dext-core-utils` in a separate PR
until then, we keep this as is?

Comment (vutran) [inline app/main/plugins.js]: Yes, let's keep as is for now.

File: package.json [hunk 2]
```diff
@@ -33,6 +33,7 @@
     "babel-eslint": "^6.1.2",
     "babel-jest": "^15.0.0",
     "babel-loader": "^6.2.5",
+    "babel-plugin-rewire": "^1.0.0",
```
Comment (adnasa) [inline package.json]: @vutran This PR now includes `babel-plugin-rewire`, 
which I use to [mock the `fs` module](https://github.com/adnasa/dext/blob/6bda0795398987327d1f32adcad0aecc78af949b/__tests__/app/main/plugins.js#L163-L171)

**Comment by adnasa** [comment]
This is mildly frustrating... the build failed, but locally everything is fine 😕 
any ideas here?

File: package.json (hunk 2 above)
Comment (vutran) [inline package.json]: We should be able to do this with `jest` as well via [manual mocks](https://facebook.github.io/jest/docs/manual-mocks.html) for this test case.

**Comment by adnasa** [comment]
> We may have to mock `fs.readdir` as well.

x

**Comment by adnasa** [comment]
that's it, I'm going to bed 😄

**Comment by adnasa** [comment]
@vutran I could not get the test to work smoothly when using `jest` mocking..
I decided to instead revert all unit-test that I've written in this branch, mostly because it blew out of scope for me.

suggestions on what to do?

**Comment by vutran** [comment]
Let's rebase and make sure it is working and merge it.

I can work on getting those `jest` mocks at a later time or we can open it to the #hacktoberfest.

**Comment by vutran** [comment]
I'll work on getting you the mocks this weekend so let's hang tight on this PR as we can rebase after.

**Comment by adnasa** [comment]
Closing since this seemed to have been addressed already

**Pull request closed without merging**